src/Persistence/PersistentObjectFactory.php

```diff
@@ -226,11 +226,13 @@ final public function create(callable|array $attributes = []): object
 
         $this->tempAfterPersist = [];
 
-        foreach ($this->afterPersist as $callback) {
-            $callback($object);
-        }
-
         if ($this->afterPersist) {
+            $attributes = $this->normalizeAttributes($attributes);
```

Comment:
not a huge fan of this:
- this is already done in `parent::create()`
- we won't pass attributes added in `beforeInstantiate` callbacks

any thought @kbond ?

Reply:
Hmm, I agree, this doesn't feel great. I guess I totally missed this part when working on 2.0...